src/compiler/scala/tools/nsc/transform/patmat/MatchTreeMaking.scala

```diff
@@ -648,7 +648,7 @@ trait MatchTreeMaking extends MatchCodeGen with Debugging {
             // avoids traversing the entire list if there are more than 3 elements
             def lengthMax3(l: List[List[TreeMaker]]): Int = l match {
               case a :: b :: c :: _ => 3
-              case cases            => cases.map {
+              case kases            => kases.map {
```

Comment:
better to shadow at `def lengthMax3(cases: List[...])` which actually represents the shadowing, and the scrutinee exclusion falls out naturally